File: CASHIQ-CASHMANAGEMENT-MICROSERVICES/CASHIQ-CASHMANAGEMENT-MANAGE-CORE/src/main/java/com/cashiq/cashmanagement/entity/Income.java
package com.cashiq.cashmanagement.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "incomes")
public class Income {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private Double amount;

    private String icon;

    private String frequency;

    private LocalDate nextPayDay;

    @Column(name = "is_fixed")
    private Boolean isFixed;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private Users user;
}

File: CASHIQ-CASHMANAGEMENT-REACT-FRONTEND/src/models/Income.ts
import { PaymentFrequency } from './PaymentFrequency';

export interface IncomeDTO {
    id?: number;
    name: string;
    amount: number;
    icon: string;
    frequency: PaymentFrequency;
    nextPayDay: string; // YYYY-MM-DD
    isFixed: boolean;
    userId?: number;
}
